### ISSUE TITLE
Is there a way to return a custom error from the handleSignUp function?

### ISSUE THREAD

**Issue by reesscot** [aws-amplify/amplify-ui#3939]
### Discussed in https://github.com/aws-amplify/amplify-ui/discussions/3119

<div type='discussions-op-text'>

<sup>Originally posted by **taphill** December  1, 2022</sup>
Hello all! I'm trying to figure out if there is a way to do the following with the react `Authenticator` component:

In order for a customer to sign up for this particular service we have, they need to provide us with a valid `customer id`. Here's what our form looks like:

<img src="https://i.ibb.co/PgdCXDj/Screen-Shot-2022-12-01-at-10-59-10-AM.png" alt="Screen-Shot-2022-12-01-at-10-59-10-AM" border="0" width="250">

Here's what I would like to do:

```javascript
// Override the default sign up function
const services= {
  async handleSignUp(formData) {
    let { username, password, attributes } = formData;

    // Take the the attributes["custom:customer_id"] and
    // make an API call to an external service to determine if
    // if the customer id is valid or not.

    if (customer_id_is_valid) {
      return Auth.signUp({
        username,
        password,
        attributes,
      });
    } else {
      // Return a custom sign up error
      // That tells the user the customer id is invalid
    }
  }
}
```

From what I can tell the `Auth.signUp` promise returns a data object of type `ISignUpResult`, so I'm assuming that's what `async handleSignUp` is expecting? But if I do something like this:

```javascript
// Override the default sign up function
const services= {
  async handleSignUp(formData) {
    let { username, password, attributes } = formData;

    // Purposefully leaving out the username to generate an error
    return Auth.signUp({
      password,
      attributes,
    });
  }
}
```

Then some type of `AuthError` is returned and I see this on the form:

<img src="https://i.ibb.co/74QTdhH/Screen-Shot-2022-12-01-at-12-05-37-PM.png" alt="Screen-Shot-2022-12-01-at-12-05-37-PM" border="0" width="250">

Basically, that's exactly what I want to happen but I want the message to be "Invalid customer id" if that is the case. Is there a way for me to accomplish that?

</div>
